### PR TITLE
op-supervisor: Include executing message info when storing logs.

### PR DESCRIPTION
**Pull request by axelKingsley**
## What
Now, when ingesting new log messages to add to the log storage, the log is checked for being an Executing Message. if it is, the data of the executing message is stored in the database.

## How
During log processing, a `DecodeExecutingMessage` is called. If the structure fits the event and ABI format, the critical components about the executing message are stored as a backend type called `ExecutingMessage`.

This stored `ExecutingMessage` is fundamentally the identifier for *some other* message to which the Executing Message points. The contents of the Executing Message are:
* Chain
* BlockNumber
* LogIndex
* Timestamp
* Hash
### Hash
Hash is a composite of two values:
* Message Hash, which is provided by the ABI and represents the hash of the referenced log payload
* Origin, or the Sending Address
These two values are concatenated and then hashed together once more to make the over-all "Log Hash". We do this to merge the Address and Payload into a single value, which makes validation simpler (the hash check now implicitly checks the address), and also to reduce the amount of storage used.

## Integration, How is this Used
* In the past... an older log has already been processed, and the LogHash has been calculated for it using the same specified logic described above.
* In the present...
* When an executing message arrives for log processing, it is discovered to be an Executing Message.
* Because it has an Executing Message, we can derive a Log Hash from it. This Log Has references some older log, and because we derive it in the same way as we did when we first processed that older log, we get the same value.
* Using the index information included in the EM (Log Index, BlockNumber, Chain), we can pull up the older log and confirm that the Log Hashes match!

## Tests
Includes new unit tests for new components, written by @ajsutton and fixed up by myself

## Thanks
Thank you to @ajsutton who put most of this PR together last month prior to some package/abi/spec changes.

### CODE REVIEW

**Review by protolambda — APPROVED** [review]

File: packages/contracts-bedrock/snapshots/abi/CrossL2Inbox.json [hunk 1]
```diff
@@ -243,4 +243,4 @@
     "name": "TargetCallFailed",
     "type": "error"
   }
-]
```
Comment (mbaxter) [inline packages/contracts-bedrock/snapshots/abi/CrossL2Inbox.json]: Looks like contract-reviewers were pinged due to this whitespace change.  Looks unrelated - can this be cut?

Comment (axelKingsley) [reply packages/contracts-bedrock/snapshots/abi/CrossL2Inbox.json]: Technically it can be cut yes, but my IDE linter thought it was important enough that it should be included whenever I save the file. I can remove it via `vi` or something.

Comment (axelKingsley) [reply packages/contracts-bedrock/snapshots/abi/CrossL2Inbox.json]:
```
truncate -s -1 packages/contracts-bedrock/snapshots/abi/CrossL2Inbox.json
```
👍

Comment (mbaxter) [reply packages/contracts-bedrock/snapshots/abi/CrossL2Inbox.json]: I think this is what was causing the [bedrock checks to fail](https://app.circleci.com/pipelines/github/ethereum-optimism/optimism/60960/workflows/13394064-5c96-4d71-84c6-9a7c4baed581/jobs/2556155) previously.
